internal/refreshtoken/refresh_token.go

```diff
@@ -76,17 +81,26 @@ func FromResource(res boundary.Resource, grantsHash []byte) *Token {
 	}
 }
 
-// Refresh refreshes a token's updated time
+// Refresh refreshes a token's updated time. It accounts for database
+// transaction inaccuracies by subtracting UpdatedTimeBuffer from the
+// provided timestamp while ensuring that the updated time is never
+// before the created time of the token.
 func (rt *Token) Refresh(updatedTime time.Time) *Token {
-	rt.UpdatedTime = updatedTime
+	rt.UpdatedTime = updatedTime.Add(UpdatedTimeBuffer)
+	if rt.UpdatedTime.Before(rt.CreatedTime) {
+		rt.UpdatedTime = rt.CreatedTime
+	}
 	return rt
 }
 
-// RefreshLastItem refreshes a token's updated time and last item
+// RefreshLastItem refreshes a token's updated time and last item.
+// It accounts for database transaction inaccuracies by subtracting
+// UpdatedTimeBuffer from the provided timestamp while ensuring that
+// the updated time is never before the created time of the token.
 func (rt *Token) RefreshLastItem(res boundary.Resource, updatedTime time.Time) *Token {
-	rt.UpdatedTime = updatedTime
 	rt.LastItemId = res.GetPublicId()
 	rt.LastItemUpdatedTime = res.GetUpdateTime().AsTime()
+	rt = rt.Refresh(updatedTime)
```

Comment:
Now that Refresh does more than a trivial amount of work, we call it from here instead of copying the logic